cli/src/commands/runTests.js

```diff
@@ -42,8 +43,22 @@ type TestGroup = {
  * structs. Each TestGroup represents a Package/PackageVersion/FlowVersion
  * directory.
  */
-async function getTestGroups(repoDirPath): Promise<Array<TestGroup>> {
-  const libDefs = await getLibDefs(repoDirPath);
+ const basePathRegex = new RegExp(`definitions${path.sep}npm${path.sep}[^${path.sep}]*${path.sep}?`);
+async function getTestGroups(repoDirPath, onlyChanged: bool = false): Promise<Array<TestGroup>> {
+  let libDefs = await getLibDefs(repoDirPath);
+  if (onlyChanged) {
+    const diff = await getDiff();
+    let changedDefs;
+    // $FlowFixMe
+    const baseDiff: string[] = diff.map(d => {
+      const match = d.match(basePathRegex);
+      if (match) {
+        return match[0];
+      }
+    }).filter(d => d != null);
+    changedDefs = baseDiff.map(d => parseRepoDirItem(d).pkgName);
```

Comment:
nit: Maybe make `changedDefs` a dict or a map so the `.filter()` below isn't exponential?